src/test/java/org/broadinstitute/hellbender/tools/PrintReadsIntegrationTest.java

```diff
@@ -39,4 +52,66 @@ public void testNoConflictPG() throws IOException {
         Assert.assertNotNull(SamReaderFactory.makeDefault().open(outFile).getFileHeader().getProgramRecord("GATK PrintReads"));
         Assert.assertNotNull(SamReaderFactory.makeDefault().open(outFile).getFileHeader().getProgramRecord("GATK PrintReads.1"));
     }
+
+    @DataProvider
+    public Object[][] getHttpPaths(){
+        final String bam = "gs://hellbender/test/resources/benchmark/CEUTrio.HiSeq.WEx.b37.NA12892.bam";
+        final String bai = "gs://hellbender/test/resources/benchmark/CEUTrio.HiSeq.WEx.b37.NA12892.bam.bai";
+        final String cram = "gs://hellbender/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.cram";
+        final String crai = "gs://hellbender/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.cram.bai";
+        final List<SimpleInterval> largeFileIntervals = Arrays.asList(new SimpleInterval("3",1_000_000, 1_000_001),
+               new SimpleInterval("3", 1_000_003, 1_000_100),
+                new SimpleInterval("20", 1_099_000, 1_100_000));
+
+        final List<SimpleInterval> cramIntervals = Arrays.asList(new SimpleInterval("20", 1_099_000, 1_100_000));
+        return new Object[][]{
+                {bucketPathToPublicHttpUrl(bam), bucketPathToPublicHttpUrl(bai), bam, bai, largeFileIntervals},
+                {toSignedUrl(bam), toSignedUrl(bai), bam, bai, largeFileIntervals},
+                {bucketPathToPublicHttpUrl(cram), bucketPathToPublicHttpUrl(crai), cram, crai, cramIntervals},
+                {toSignedUrl(cram), toSignedUrl(crai), cram, crai, cramIntervals}
+
+        };
+    }
+
+    @Test(groups = {"cloud", "bucket"}, dataProvider = "getHttpPaths")
+    public void testHttpPaths(String reads, String index, String nonHttpReads, String nonHttpIndex, List<SimpleInterval> intervals) throws IOException {
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        final File out = createTempFile("out", ".bam");
+        System.out.println("reading http");
+        args.addInput(reads)
+                .add(StandardArgumentDefinitions.CLOUD_PREFETCH_BUFFER_LONG_NAME, 1)
+                .add(StandardArgumentDefinitions.CLOUD_INDEX_PREFETCH_BUFFER_LONG_NAME, 1)
```

Comment:
What happens if you run with the default prefetch settings?

Reply:
It downloads 40mb chunks around the 1kb spots it's looking for.